packages/core/src/amazonq/commons/controllers/contentController.ts

```diff
@@ -4,9 +4,26 @@
  */
 
 import * as vscode from 'vscode'
+import fs from 'fs-extra'
+import path from 'path'
 import { Position, TextEditor, window } from 'vscode'
 import { getLogger } from '../../../shared/logger'
+import { AMAZON_Q_DIFF_SCHEME, amazonQTabSuffix } from '../../../shared/constants'
+import { disposeOnEditorClose } from '../../../shared/utilities/editorUtilities'
+import {
+    applyChanges,
+    createTempFileForDiff,
+    getSelectionFromRange,
+} from '../../../shared/utilities/textDocumentUtilities'
+import { extractFileAndCodeSelectionFromMessage, getIndentedCode } from '../../../shared'
 
+class ContentProvider implements vscode.TextDocumentContentProvider {
+    constructor(private uri: vscode.Uri) {}
+
+    provideTextDocumentContent(_uri: vscode.Uri) {
+        return fs.readFileSync(this.uri.fsPath, 'utf-8')
```

Comment:
Unless this needs to be synchronous, do not use fs-extra. The correct fs module is in our shared folder

Reply:
Looking at the api, this method can be async. The type eventually resolves to a possible "Thenable". So you can use our built in fs module